src/components/ItemsList.js

```diff
@@ -1,32 +1,47 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import firebase from '../lib/firebase';
-import { useCollectionData } from 'react-firebase-hooks/firestore';
+
+const userToken = localStorage.getItem('token');
+
+const db = firebase.firestore().collection('shopping_list');
 
 const ItemsList = () => {
-  const [itemsList, loading, error] = useCollectionData(
-    firebase.firestore().collection('items'),
-    {
-      snapshotListenOptions: { includeMetadataChanges: true },
-    },
-  );
-  // useCollectionData extracts firebase.firestore.QuerySnapshot.docs while useCollection
-  // returns firebase.firestore.QuerySnapshot itself.
+  const [items, setItems] = useState('');
+
+  useEffect(() => {
+    let mounted = true;
+    db.where('token', '==', userToken)
+      .get()
+      .then((data) => {
+        if (data.docs.length) {
+          let itemsList = data.docs[0]
+            .data()
+            .items.map((element) => element.shoppingListItemName);
+          if (mounted) {
+            setItems(itemsList);
+          }
+        }
+      });
+    return () => (mounted = false);
+  });
   return (
     <div>
       <h2>New List</h2>
-      {loading === true ? <p> Loading... </p> : null}
-      {error === undefined ? null : <p> An error has occurred... </p>}
-      {itemsList === undefined ? null : (
-        <ol>
-          {itemsList.map((itemObject, index) => (
-            <li key={itemObject.item + index}>
-              <div>
-                {itemObject.item},{itemObject.how_much}
-              </div>
-            </li>
-          ))}
-        </ol>
-      )}
+      <div>
+        <ul>
+          {items.length ? (
+            items.map((val, index) => {
+              return (
+                <li key={index}>
+                  <p>{val}</p>
+                </li>
+              );
+            })
+          ) : (
+            <p>Loading ...</p>
+          )}
+        </ul>
+      </div>
     </div>
```

Comment:
I love how you check if there are items in the array and then render each individual item. 

I have one suggestion:
Right now if the `shopping_list` collection is empty we always render `<p>Loading...</p>`. I think it would be better to 
1. First check if we are fetching the data from the database. While it is fetching render `Loading...`. (if loading is true, render `Loading...`).
2. When the data is fetched, show the items.
3. Otherwise you can show either an empty page or render `List is empty`. 

In other words, what I'm trying to suggest is very similar to what you had before: 

<img width="990" alt="Screen Shot 2021-01-16 at 3 59 52 PM" src="https://user-images.githubusercontent.com/45564632/104811303-7ac8fc80-57c8-11eb-86b5-b43893932233.png">

Reply:
ok. I have updated it to not show loading at all.